### ISSUE TITLE
Support OIDC auth to cloud providers

### ISSUE THREAD

**Issue by ppennanen** [dependabot/dependabot-core#6580]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature description

It would be great to utilise [GitHub OpenID Connect](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect) for dependabot, to avoid the need to store static credentials in GitHub secrets.

This is particularly important for the `docker` ecosystem, where it is common that private container images are stored in cloud provider container registries (e.g. ACR, ECR and GCR).

**Comment by jeffwidman** [comment]
This is a great idea!

Unfortunately, implementing that will take a bit of work--at least for the Dependabot service that GitHub runs, so not something we're likely to get to for a while. But it is something I could see us eventually building out.

In the meantime, if you want a workaround, the Dependabot secrets can be [rotated via an API](https://docs.github.com/en/rest/dependabot/secrets?apiVersion=2022-11-28). So you could script the rotation on your desired frequency... for example, if Dependabot runs weekly, you can script the token to rotate the day before with a 24 hour expiration, so it expires right after Dependabot runs. There's even a community-maintained [GitHub Action for it](https://github.com/marketplace/actions/set-action-dependabot-secret).

**Comment by ppennanen** [comment]
Thank you for the feedback and workaround!

**Comment by joelbyford** [comment]
+1 on this issue.

**Comment by larhauga** [comment]
@jeffwidman is there any plans to implement this? As dependabot is transitioning to running on actions workflows, a _new_ possibility would could be to allow for custom extra workflow steps that could enable custom authentication. The github OIDC could then be used in a custom step to gain access to temporary roles, while exporting the tokens to dependabot.

**Comment by jluque0101** [comment]
+1 on this as it forces us to lower our security posture for our setup by using static credentials or alternatively implement more complex solutions to go around it.

**Comment by OuranosSkia** [comment]
Also +1 on this - We use OIDC for our normal Github Actions at my company, but then having to embed auth credentials in dependabot secrets to allow dependabot to run properly makes the gains from having OIDC in the workflows a little moot. It would be amazing to be able to have this all the way through.

**Comment by dchien234** [comment]
+1 for this. I think it's really worth it, if you want to encourage `dependabot` usage in enterprise environment, where OIDC usage is kind of a must.

**Comment by austinhallett** [comment]
+1 For this

**Comment by lplanas** [comment]
I saw this announcement in the official GitHub blog, so it seems our prayers have finally been listened <https://github.blog/changelog/2026-02-03-dependabot-now-supports-oidc-authentication/>

**Comment by pealtrufo** [comment]
> I saw this announcement in the official GitHub blog, so it seems our prayers have finally been listened https://github.blog/changelog/2026-02-03-dependabot-now-supports-oidc-authentication/

I have been looking for documentation on how to use it in the docs but couldn't find anything? Is this not documented yet?

**Comment by lplanas** [comment]
> > I saw this announcement in the official GitHub blog, so it seems our prayers have finally been listened https://github.blog/changelog/2026-02-03-dependabot-now-supports-oidc-authentication/
> 
> I have been looking for documentation on how to use it in the docs but couldn't find anything? Is this not documented yet?

Yeah, I have been searching for it too, with no luck, some AI agents are heavily hallucinating, but one of them pointed towards the implementation in <https://github.com/dependabot/proxy/blob/main/internal/oidc/actions_oidc.go>

**Comment by OuranosSkia** [comment]
Per the docs you linked, it says:
> To configure OIDC authentication, you need to specify `tenant-id` and `client-id` instead of `username` and `password` in your registry configuration.

I haven't tried it yet (as I am on Enterprise and will likely need to wait for my enterprise to update to a version that will have this), but I would assume you would define a registry like this:
```yaml
registries:
  gradle-artifactory:
    type: maven-repository
    url: https://acme.jfrog.io/artifactory/my-gradle-registry
    tenant-id: my_tenant_id
    client-id: my_client_id
```

Glad that this is finally out and excited to eventually be able to integrate it!

**Comment by pealtrufo** [comment]
Yes, they've just added a new section for OIDC in the documentation: 

https://docs.github.com/en/code-security/how-tos/secure-your-supply-chain/manage-your-dependency-security/configuring-access-to-private-registries-for-dependabot#using-oidc-for-authentication

**Comment by Ferroman** [comment]
> Yes, they've just added a new section for OIDC in the documentation:
> 
> https://docs.github.com/en/code-security/how-tos/secure-your-supply-chain/manage-your-dependency-security/configuring-access-to-private-registries-for-dependabot#using-oidc-for-authentication

but not for AWS ECR
